src/components/forms/MemoryLimitAvailable.tsx

```diff
@@ -1,59 +1,149 @@
 import type { FC } from "react";
 import { useNotify, Spinner } from "@canonical/react-components";
-import { useQuery } from "@tanstack/react-query";
-import { fetchResources } from "api/server";
-import { queryKeys } from "util/queryKeys";
 import { humanFileSize } from "util/helpers";
 import { limitToBytes } from "util/limits";
 import type { LxdProject } from "types/project";
-import { useServerEntitlements } from "util/entitlements/server";
+import type { LxdResources } from "types/resources";
+import { Icon } from "@canonical/react-components";
+import type { InstanceAndProfileFormikProps } from "./instanceAndProfileFormValues";
+import type { CreateInstanceFormValues } from "pages/instances/CreateInstance";
+import type { EditInstanceFormValues } from "pages/instances/EditInstance";
+import { CLUSTER_GROUP_PREFIX } from "util/instances";
+import ResourceLink from "components/ResourceLink";
+import { useIsClustered } from "context/useIsClustered";
+import { useResources } from "context/useResources";
 
+interface MemoryLimit {
+  min: number;
+  max: number;
+  source: "project" | "cluster-member" | "";
+}
 interface Props {
+  formik?: InstanceAndProfileFormikProps;
   project?: LxdProject;
 }
 
-const MemoryLimitAvailable: FC<Props> = ({ project }) => {
+const MemoryLimitAvailable: FC<Props> = ({ formik, project }) => {
   const notify = useNotify();
-  const { canViewResources } = useServerEntitlements();
+  const isClustered = useIsClustered();
+  const values = formik?.values;
+  const projectResourceMemory = limitToBytes(project?.config["limits.memory"]);
+  const currentServerInCluster = (() => {
+    if (!isClustered) {
+      return undefined;
+    }
+    const createTarget =
+      (values as CreateInstanceFormValues)?.target ?? undefined;
+    if (createTarget && !createTarget.startsWith(CLUSTER_GROUP_PREFIX)) {
+      return createTarget;
+    }
+    const location = (values as EditInstanceFormValues)?.location ?? undefined;
+    return location || undefined;
+  })();
 
   const {
     data: resources,
-    error,
-    isLoading,
-  } = useQuery({
-    queryKey: [queryKeys.resources],
-    queryFn: async () => fetchResources(),
-    enabled: canViewResources(),
-  });
+    error: error,
+    isLoading: isLoading,
+  } = useResources(currentServerInCluster);
 
   if (isLoading) {
     return <Spinner className="u-loader" text="Loading resources..." />;
   }
-
   if (error) {
     notify.failure("Loading resources failed", error);
+    return null;
   }
 
-  const getAvailableMemory = () => {
-    if (!project?.config["limits.memory"]) {
-      return resources?.memory.total;
+  const getSourceQualifier = () => {
```

Comment:
```suggestion
  const getSourceQualifier = (source: string) => {
```
I guess without passing it in, this will always return empty.